library/std/src/io/impls.rs

```diff
@@ -306,6 +306,9 @@ impl BufRead for &[u8] {
 ///
 /// Note that writing updates the slice to point to the yet unwritten part.
 /// The slice will be empty when it has been completely overwritten.
+///
+/// If the amount to be written exceeds the size of the slice,
+/// an error of kind `ErrorKind::WriteZero` is signaled.
```

Comment:
That's only the case for `write_all`. `write` and `write_vectored` just return `Ok(n)` with the number of bytes written (even if that's zero).

Reply:
You are right, will add a commit to fix.